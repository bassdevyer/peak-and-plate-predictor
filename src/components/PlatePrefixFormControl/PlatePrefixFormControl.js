import React from 'react'
import './style.css'
import { FormControl } from 'react-bootstrap'

const _onPlatePrefixKeyPress = e => {
  const regExp = /[a-zA-Z]+/g
  if (!regExp.test(e.key)) {
    e.preventDefault()
  }
}

const PlatePrefixFormControl = (props) => {
  return (
    <FormControl
      {...props}/>
  )
}

PlatePrefixFormControl.defaultProps = {
  className: 'platePrefixFormControl',
  type: 'text',
  maxLength: 3,
  onKeyPress: (e) => _onPlatePrefixKeyPress(
    e),
}

export default PlatePrefixFormControl