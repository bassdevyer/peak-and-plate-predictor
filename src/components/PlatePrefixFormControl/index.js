import PlatePrefixFormControl from './PlatePrefixFormControl'

export default PlatePrefixFormControl