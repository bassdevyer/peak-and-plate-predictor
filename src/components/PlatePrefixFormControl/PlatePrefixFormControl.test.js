import React from 'react'
import PlatePrefixFormControl from '../PlatePrefixFormControl'
import { create } from 'react-test-renderer'

describe('Component default rendering', () => {
  it('Renders component correctly', () => {
    const component = create(
      <PlatePrefixFormControl/>,
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Testing props', () => {
  it('Sets default props', () => {
    const props = {
      className: 'platePrefixFormControl',
      type: 'text',
      maxLength: 3,
    }
    expect(PlatePrefixFormControl.defaultProps.className)
      .toEqual(props.className)
    expect(PlatePrefixFormControl.defaultProps.type)
      .toEqual(props.type)
    expect(PlatePrefixFormControl.defaultProps.maxLength)
      .toEqual(props.maxLength)
    expect(PlatePrefixFormControl.defaultProps.onKeyPress)
      .toBeDefined()
  })

  it('Sets external props', () => {
    const props = {
      value: '',
      onChange: jest.fn(),
    }
    const component = create(
      <PlatePrefixFormControl/>,
    )
    expect(PlatePrefixFormControl.props.className)
      .toEqual(props.className)
    expect(PlatePrefixFormControl.defaultProps.type)
      .toEqual(props.type)
    expect(PlatePrefixFormControl.defaultProps.maxLength)
      .toEqual(props.maxLength)
    expect(PlatePrefixFormControl.defaultProps.onKeyPress)
      .toBeDefined()
  })

})

describe('Testing only-letters acceptance', () => {
  it('Accepts only letters', () => {

  })
})