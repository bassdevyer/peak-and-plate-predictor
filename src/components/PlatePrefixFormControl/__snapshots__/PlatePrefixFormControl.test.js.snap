// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component default rendering Renders component correctly 1`] = `
<input
  className="platePrefixFormControl form-control"
  maxLength={3}
  onKeyPress={[Function]}
  type="text"
/>
`;
